core/primitives/big_int.hpp

```diff
@@ -21,6 +21,38 @@ namespace fc::primitives {
     using cpp_int::cpp_int;
   };
 
+  static inline const BigInt operator+(const BigInt &lhs, const BigInt &rhs) {
+    return static_cast<cpp_int>(lhs) + static_cast<cpp_int>(rhs);
+  }
+
+  static inline const BigInt operator-(const BigInt &lhs, const BigInt &rhs) {
+    return static_cast<cpp_int>(lhs) - static_cast<cpp_int>(rhs);
+  }
+
+  static inline bool operator>=(const BigInt &lhs, const BigInt &rhs) {
+    return static_cast<cpp_int>(lhs) >= static_cast<cpp_int>(rhs);
+  }
+
+  static inline bool operator>(const BigInt &lhs, const BigInt &rhs) {
+    return static_cast<cpp_int>(lhs) > static_cast<cpp_int>(rhs);
+  }
+
+  static inline bool operator<=(const BigInt &lhs, const BigInt &rhs) {
+    return static_cast<cpp_int>(lhs) <= static_cast<cpp_int>(rhs);
+  }
+
+  static inline bool operator<=(const BigInt &lhs, int rhs) {
+    return static_cast<cpp_int>(lhs) <= rhs;
+  }
+
+  static inline bool operator<(const BigInt &lhs, const BigInt &rhs) {
+    return static_cast<cpp_int>(lhs) < static_cast<cpp_int>(rhs);
```

Comment:
newbee cpp question: is that a bad practice to reuse the next method here or is it impossible?

Reply:
hmm,, I don't know. But it adds dependency between them. Error in one will lead to incorrect work of both

Reply:
generally it is better to have the only place to contain logic. So each either works correctly or not